resources/js/components/people/gifts/CreateGift.vue

```diff
@@ -0,0 +1,378 @@
+<style scoped>
+    .photo {
+        height: 200px;
+    }
+</style>
+
+<template>
+  <div>
+    <!-- Add a gift -->
+    <transition name="fade">
+      <div class="ba br3 mb3 pa3 b--black-40">
+        <div class="pb3 mb3 flex-ns b--gray-monica">
+          <!-- STATUS -->
+          <form-radio
+            :id="'status_idea'"
+            v-model="newGift.status"
+            :name="'status'"
+            :required="true"
+            :value="'idea'"
+            :color="'success'"
+            :full-class="'p-default p-fill p-curve'"
+          >
+            {{ $t('people.gifts_add_gift_idea') }}
+          </form-radio>
+
+          <form-radio
+            :id="'status_offered'"
+            v-model="newGift.status"
+            :name="'status'"
+            :required="true"
+            :value="'offered'"
+            :color="'info'"
+            :full-class="'p-default p-fill p-curve'"
+          >
+            {{ $t('people.gifts_add_gift_already_offered') }}
+          </form-radio>
+
+          <form-radio
+            :id="'status_received'"
+            v-model="newGift.status"
+            :name="'status'"
+            :required="true"
+            :value="'received'"
+            :color="'warning'"
+            :full-class="'p-default p-fill p-curve'"
+          >
+            {{ $t('people.gifts_add_gift_received') }}
+          </form-radio>
+        </div>
+
+        <div class="dt dt--fixed pb3 mb3 mb0-ns bb b--gray-monica">
+          <!-- NAME -->
+          <form-input
+            :id="'name'"
+            v-model="newGift.name"
+            :input-type="'text'"
+            :maxlength="255"
+            :required="true"
+            :class="'dtc pr2'"
+            :title="$t('people.gifts_add_gift_name')"
+            @submit="store"
+          />
+        </div>
+
+        <!-- ADDITIONAL FIELDS -->
+        <div v-show="!displayComment || !displayUrl || !displayAmount || !(displayRecipient && familyContacts.length == 0) || !(displayUpload && reachLimit)" class="bb b--gray-monica pv3 mb3">
```

Comment:
There are a lot of conditions here. It would probably be better to have a function instead, don't you think?